### PR TITLE
Feature/calendar links and events

### PR DESCRIPTION
**Pull request by doingandlearning**
This needs to be developed in line with the new time and duration work being done in parallel. 

I'd initially added filename to the YAML but realise now that nuxt/content gives me access to the path and so this is unnecessary and I'll revert.

### CODE REVIEW

File: components/SessionPage.vue [hunk 1]
```diff
@@ -167,6 +182,48 @@ export default {
...
+        if (parameters.hasOwnProperty(key) && parameters[key]) {
+          url += `&${key}=${parameters[key]}`;
+        }
+      }
+
+      return url;
+    },
+  },
+  methods: {
+    isValidDate(d) {
```
Comment (phivk) [inline components/SessionPage.vue:217]: what is `isValidDate()` doing here?

File: content/mozfest/sessions/data-science-meets-law-learning-responsible-ai-together.md [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
   text: learn.responsibly.ai
   href: http://learn.responsibly.ai/
+filename: data-science-meets-law-learning-responsible-ai-together
 ---
+
 Responsible AI challenges require a multidisciplinary approach, in which tech and law are natural partners.
+
 <!--more-->
+
+<<<<<<< HEAD
```
Comment (phivk) [inline content/mozfest/sessions/data-science-meets-law-learning-responsible-ai-together.md:26]: merge conflict artifact?

File: generateCalendarEvents.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          eventDate.getFullYear(),
+          eventDate.getMonth() + 1, // Months are zero indexed
+          eventDate.getDate(),
+          eventDate.getHours(),
+          eventDate.getMinutes()
+        ],
+        duration: getDuration(frontMatter.durationInMinutes),
+        title: frontMatter.title,
+        description: `A ${frontMatter.type} event.`,
+        location: "On Zoom - find link on the #skillsharing channel on Slack"
```
Comment (phivk) [inline generateCalendarEvents.js:43]: this text is specific to the MozFest community. how will the script work with SU?

Comment (phivk) [reply generateCalendarEvents.js:43]: I see the script goes per community and we are commenting out SU now anyway. 
no need to fix right now, but let's think about this for the future. I think this CTA could live in the communities.json config...

File: static/mozfest/technological-suffering- and-how-mindfulness-can-help.ics [hunk 4]
```diff
@@ -0,0 +1,16 @@
+BEGIN:VCALENDAR
```
Comment (phivk) [inline static/mozfest/technological-suffering- and-how-mindfulness-can-help.ics:1]: the file name seems to have an extra space and is therefore a duplicate of the next file

**Review by phivk — CHANGES_REQUESTED** [review]
nice approach! see my comments in line :)

**Comment by phivk** [comment]
@doingandlearning hey, just checking if we can make progress on this?

File: generateCalendarEvents.js (hunk 3 above)
Comment (doingandlearning) [reply generateCalendarEvents.js:43]: Maybe this could be {frontmatter.location} with a fallback to a set string.

**Review by doingandlearning — COMMENTED** [review]
I am very sorry for how long this has lay dormant!

I've tidied it up a bit now, merged current master in and acted on your comments.

File: generateCalendarEvents.js (hunk 3 above)
Comment (phivk) [reply generateCalendarEvents.js:43]: yes, that seems like an elegant solution. 
could you make that adjustment before I merge?

**Review by phivk — CHANGES_REQUESTED** [review]
looks good. I like your proposal to get Calendar location from frontmatter and have a fallback string in place. To give the fallback string some extra longevity, we could use "Find location details on the community skill sharing channel".
Could you make this adjustment before I merge?

**Comment by doingandlearning** [comment]
Cool! I've set up with three progressively less specific options:
- Front matter
- Default location for a given community (using the community.js object)
- A set string directing people back to the channel (as mentioned in your comment)

**Review by phivk — DISMISSED** [review]

**Comment by phivk** [comment]
left: local test event, right: actual live event
notice the 1 hour time difference

23AAEF4E-B920-45FB-8C3A-1BC82A0BEF41-878-00005CC6A83B633A

the generated .ics files have the same issue:

7FF4423E-44B0-4A2C-AECF-23226677F9B5-878-00005CFAA2DE6A28

**Comment by phivk** [comment]
might be helpful to look at the SU ics files I created manually, e.g. https://github.com/phivk/gradu.al/blob/a71cdc7772104eb8813c27c7be1bc7d9d0d23027/static/storytellersunited/201110-newsletters.ics

**Comment by doingandlearning** [comment]
Interesting … all of the events are using the TZ in the format `'2021-06-14T17:00+01:00'` - presumably because of CET timings - but I guess in summer time it’s actually +2. 

The JS Date function is taking the time and converting it into UTC (`2021-06-14T16:00:00.000Z`) and using that to create the ICS event. 

I dislike timezones :) 

Could you send me a copy of your test event (Bart and Beers) - can't see it in the repo?

**Comment by phivk** [comment]
> I dislike timezones :)

which developer doesn't ;)

> Could you send me a copy of your test event (Bart and Beers) - can't see it in the repo?

see slack :)

**Comment by phivk** [comment]
@doingandlearning I think this PR got closed accidentally (maybe by me renaming `master` to `main`?).
do you know a way to reopen?
alternatively we could open a new PR to merge from this feature branch to `develop`.

**Comment by doingandlearning** [comment]
Trying to remove dependency on anything that might be swapped out, I've got the ICS files being generated on page build.

**Comment by doingandlearning** [comment]
To test: 
- Select an upcoming event
- Click to download the ICS 
- Upload the file or copy the contents to a validator like this one (https://icalendar.org/validator.html#results)
- Confirm that the details match

**Pull request closed without merging**